### ISSUE TITLE
pwtorrents.net torrents not recognised as private

### ISSUE THREAD

**Issue by ADz-83** [Novik/ruTorrent#2385]
### Please complete the following tasks.

- [X] Web browser cache cleared
- [X] Link provided to install script if applicable
- [X] Web browser, ruTorrent, PHP and OS version provided

### Tell us about your environment

Firefox v104.0.2
ruTorrent v4.0-beta
Debian 11

### Tell us how you installed ruTorrent

Quickbox Pro
https://quickbox.io

### Describe the bug

pwtorrents.net seedbox option/tracker announce isn't showing as private so gets limited by ratio rules etc.

Full tracker announce is
http://pwtorrents.net/seedbox.php

There is no passkey as it uses pre-defined IP address in profile.

### Steps to reproduce

1. Set ratio rules/extratio/throttle plugins to limit public tracker
2. Add torrent from pwtorrents.net using Seedbox IP option (IP added to profile, use seedbox button on site to grab .torrent)

### Expected behavior

Recognised as Private not Public

**Comment by stickz** [comment]
This is related to the **plimits** plugin. Without the plugin, ruTorrent doesn't know if the tracker is public or private. As a result, you have no way of limiting public trackers in the first place.

Since you are running **ruTorrent v4.0** with this plugin, you will need to update the `trackers.lst` file. [http://pwtorrents.net/](http://pwtorrents.net/) (or anther domain resolving to the same IP address) is most likely listed as public in this file. More information is available here on the third-party GitHub repository for this plugin. https://github.com/Micdu70/rutorrent-plimits

**Comment by ADz-83** [comment]
> This is related to the **plimits** plugin. Without the plugin, ruTorrent doesn't know if the tracker is public or private. As a result, you have no way of limiting public trackers in the first place.
> 
> Since you are running **ruTorrent v4.0** with this plugin, you will need to update the `trackers.lst` file. http://pwtorrents.net/ (or anther domain resolving to the same IP address) is most likely listed as public in this file. More information is available here on the third-party GitHub repository for this plugin. https://github.com/Micdu70/rutorrent-plimits

This has absolutely **zero** to do with that plugin because I'm **not** using it and yes ruorrent does know if a torrent is public or private ad yes you do have ways of limiting public trackers via he ratio rules and throttle plugin as I have already mentioned.

**Comment by Novik** [comment]
> Full tracker announce is
> http://pwtorrents.net/seedbox.php
> There is no passkey as it uses pre-defined IP address in profile.

This is a normal behavior and can't be fixed. 

@stickz see https://github.com/Novik/ruTorrent/blob/master/plugins/extratio/rules.php#L33

**Comment by stickz** [comment]
@Novik Yeah I saw the regex patterns, to determinate if a tracker is private. `pwtorrents.net` doesn't match any of these patterns. They whitelist by IP Address instead of using passkeys.

The next best option is to add a new item to `theContextMenu` for trackers. This would allow the user to change the tracker type themselves to public or private. It makes sense to cache the `preg_match` results for trackers using the `rCache` class anyways. They become problematic when run on thousands of trackers at once. It's only a couple more steps to override these values.

**Comment by Antorell** [comment]
I had a similar issue the other day where rutorrent randomly flagged a public tracker as private. Can't rutorrent get the torrent's file private flag to find out if a torrent is public or private instead of using regexes?

I use PluginExtRatio, not plimit.

**Comment by ILLYAG** [comment]
> > Full tracker announce is
> > http://pwtorrents.net/seedbox.php
> > There is no passkey as it uses pre-defined IP address in profile.
> 
> This is a normal behavior and can't be fixed.
> 
> @stickz see https://github.com/Novik/ruTorrent/blob/master/plugins/extratio/rules.php#L33

In past you have added exceptions so that some trackers are set to private in rutorrent.  Could you not do that here for this tracker?

**Comment by ILLYAG** [comment]
> @Novik Yeah I saw the regex patterns, to determinate if a tracker is private. `pwtorrents.net` doesn't match any of these patterns. They whitelist by IP Address instead of using passkeys.
> 
> The next best option is to add a new item to `theContextMenu` for trackers. This would allow the user to change the tracker type themselves to public or private. It makes sense to cache the `preg_match` results for trackers using the `rCache` class anyways. They become problematic when run on thousands of trackers at once. It's only a couple more steps to override these values.

It's only their seedbox option/torrent button that uses IP.  Their regular button uses passkey.

**Comment by stickz** [comment]
> I had a similar issue the other day where rutorrent randomly flagged a public tracker as private 

Please create a separate bug report. I require more specific details. Follow all the steps and include a screenshot.

> Can't rutorrent get the torrent's file private flag to find out if a torrent is public or private instead of using regexes?

No, that is not a good solution. The torrent's private flag has nothing to do with it having public or private trackers.

> In past you have added exceptions so that some trackers are set to private in rutorrent. Could you not do that here for this tracker?

I just mentioned allowing the user to right click on the tracker and set it to private themselves. This is more much convenient. If you want to create an issue report, ask us to do it and update your ruTorrent version instead please let me know.

> It's only their seedbox option/torrent button that uses IP. Their regular button uses passkey.

This doesn't matter. Once the user sets the whole website to private, it will work properly for both these options.

**Comment by Antorell** [comment]
I'm not going to create another issue when the issue is obviously because of the same regex.

>  Can't rutorrent get the torrent's file private flag to find out if a torrent is public or private instead of using regexes?
> 
> No, that is not a good solution. The torrent's private flag has nothing to do with it having public or private trackers.

lol? The private flag disable DHT PEX and the like, ever seen a private tracker with DHT/PEX? If you mean because you can add public trackers to your private tracker, and ext ratio wouldn't find that. What's the point of being in a public tracker if it's to add public trackers to every torrent you download from that private tracker? That's a solution for an niche use.

**Comment by stickz** [comment]
@ADz-83 I'm working on a solution to your problem. I locked this conversation temporarily, as the issue report is going downhill.

**Comment by xirvik** [comment]
Closing, too old - reopen if needed or (better) create a new one with updated info.
